flink-end-to-end-tests/test-scripts/kafka-common.sh

```diff
@@ -59,13 +59,29 @@ function setup_confluent_dist {
   sed -i -e "s#listeners=http://0.0.0.0:8081#listeners=http://0.0.0.0:${SCHEMA_REGISTRY_PORT}#" $CONFLUENT_DIR/etc/schema-registry/schema-registry.properties
 }
 
+function wait_for_zookeeper_running {
+  while $KAFKA_DIR/bin/zookeeper-shell.sh localhost:2181 get /testnonexistent 2>&1 | grep -q "Connection refused"; do
+    echo "waiting for zookeeper to start."
+    sleep 1
+  done
```

Comment:
Shouldn't we add a counter here? Otherwise we'd create an infinite loop if the zookeeper startup fails. The old implementation would have ended up exiting in the [while loop](https://github.com/apache/flink/pull/13248/files#diff-3a25c674bfb7f722ef91f6b8630e56f4R99) in such a case. I think, we don't reach that one in case of a missing ZooKeeper anymore...

Reply:
Yes, that's a good point. I'll add one.